Makefile

```
@@ -299,14 +299,17 @@ debug-jupyter-notebook: build-release-image-jupyter ## Debug jupyter notebook
 	docker run \
 		-it \
 		--rm \
+		--user=root \
 		--publish="$(jupyter_notebook_port):8888" \
 		--volume="$(CURDIR)/bindings/python/docs:/home/jovyan/docs" \
 		--volume="$(CURDIR)/tutorials/python/notebooks:/home/jovyan/tutorials" \
 		--volume="$(CURDIR)/lib/libopen-space-toolkit-$(project_name).so.$(project_major_version):/opt/conda/lib/python$(jupyter_python_version)/site-packages/ostk/$(project_name)/libopen-space-toolkit-$(project_name).so.$(project_major_version):ro" \
 		--volume="$(CURDIR)/lib/$(jupyter_project_name_python_shared_object).so:/opt/conda/lib/python$(jupyter_python_version)/site-packages/ostk/$(project_name)/$(jupyter_project_name_python_shared_object).so:ro" \
 		--workdir="/home/jovyan" \
 		$(docker_release_image_jupyter_repository):$(docker_image_version) \
-		bash -c "start-notebook.sh --ServerApp.token=''"
+		bash -c "chown -R jovyan:users /home/jovyan ; start-notebook.sh --ServerApp.token=''"
+
+	bash -c "sudo chown -R $(shell id -u):$(shell id -g) $(CURDIR)/bindings/python/docs"
```

```suggestion
		bash -c "chmod -R g+w /home/jovyan ; start-notebook.sh --ServerApp.token=''"
```


Not sure how to replicate your permissions situation, but you may be able to do something like this (+ the group add above) to avoid having a cleanup step.

Not tested, but the intent is to add the container user to the same group as the shell user, and just give group write permissions.

---

@kyle-cochran This suggestion along with the --group-add doesn't seem to work for my purposes. I'm not sure exactly what you were intending with it.

---

(this is pedantic, if it doesn't work, feel free to resolve this comment)

---

Yes correct that's the issue at hand, but your suggestion didn't seem to give jovyan write access to the jupyter notebooks inside the jovyan dir

---

Hmm.. the idea is to add group write access to the folder so that any user in the same group as the owner can write, then add jovyan to the same group as the owner. (rather than changing ownership)

I totally didn't test it though, so if something is fishy feel free to resolve this and 🚢

---

Yeah I get the idea, its just that the current implementation you proposed doesn't seem to work. So if you want to try it real quick to see if you can get your implementation to work (run the make recipe with your changes and try to save an edited jupyter notebook from the browser) then go for it. Otherwise I think its probs good enough for now. @kyle-cochran

---

resolved over slack